src/sentry/dynamic_sampling/latest_release_booster.py

```diff
@@ -84,151 +105,206 @@ def _get_releases_models(self, limit: int) -> Dict[int, Release]:
         }
 
 
-def _get_environment_cache_key_suffix(environment: Optional[str]) -> str:
-    return f"{ENVIRONMENT_SEPARATOR}{environment}" if environment else ""
+@dataclass(frozen=True)
+class LatestReleaseParams:
+    project_id: int
+    release_id: int
+    environment: Optional[str]
 
 
-def get_redis_client_for_ds() -> Any:
-    cluster_key = getattr(settings, "SENTRY_DYNAMIC_SAMPLING_RULES_REDIS_CLUSTER", "default")
-    return redis.redis_clusters.get(cluster_key)
+@dataclass(frozen=True)
+class ObservedRelease:
```

Comment:
```suggestion
class BaseRelease:
    environment: str
    release_id: int
```

`class LatestRelease(Release)`
`ObservedRelease(Release)`
may be?